src/test/java/jenkins/plugins/http_request/HttpRequestTest.java

```diff
@@ -0,0 +1,617 @@
+package jenkins.plugins.http_request;
+
+import hudson.model.Cause.UserIdCause;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.ParametersAction;
+import hudson.model.Result;
+import hudson.model.StringParameterValue;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import jenkins.plugins.http_request.auth.BasicDigestAuthentication;
+import jenkins.plugins.http_request.auth.FormAuthentication;
+import jenkins.plugins.http_request.util.NameValuePair;
+import jenkins.plugins.http_request.util.RequestAction;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.http.HttpHost;
+
+import org.junit.Test;
+import static org.junit.Assert.assertTrue;
+
+public class HttpRequestTest extends HttpRequestTestBase {
+
+    @Test
+    public void canBeSerialized() throws Exception {
+        HttpRequest httpRequest = new HttpRequest("http://localhost");
+        httpRequest.readResolve();
```

Comment:
Not sure what this is really testing now, but if you do keep a `readResolve` method for some reason, you would want to do something like serialize and then deserialize it via `XStream2`. Probably can just delete this test.